pkg-config.cmake

```diff
@@ -761,9 +761,16 @@ MACRO(PKG_CONFIG_USE_LCOMPILE_DEPENDENCY TARGET PREFIX)
 
   # Include/libraries paths seems to be filtered on Linux, add paths
   # again.
-  INCLUDE_DIRECTORIES(SYSTEM ${${PREFIX}_INCLUDE_DIRS})
+  IF(NOT NO_INCLUDE_SYSTEM)
+    SET(INCLUDE_DIRECTORIES_VAR "SYSTEM ${${PREFIX}_INCLUDE_DIRS}")
+    SET(DEBUG_INCLUDE_DIRECTORIES_VAR "SYSTEM ${${PREFIX}_DEBUG_INCLUDE_DIRS}")
+  ELSE(NOT NO_INCLUDE_SYSTEM)
+    SET(INCLUDE_DIRECTORIES_VAR "${${PREFIX}_INCLUDE_DIRS}")
+    SET(DEBUG_INCLUDE_DIRECTORIES_VAR "${${PREFIX}_DEBUG_INCLUDE_DIRS}")    
+  ENDIF(NOT NO_INCLUDE_SYSTEM)
```

Comment:
I tested this on a simple project and this is broken. Here the CMakeLists.txt file where the call happens is `/tmp/foo` and this ends up in having `-I"/tmp/foo/SYSTEM /some/path"`

AFAIK, the proper way to do it is not to introduce an intermediary variable but call `include_directories` with/without the `SYSTEM` option.